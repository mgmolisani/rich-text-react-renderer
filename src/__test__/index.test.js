import { BLOCKS, INLINES, MARKS } from '@contentful/rich-text-types';
import { documentToReactTree } from '../';
import React from 'react';

import {
  embeddedEntryDoc,
  headingDoc,
  hrDoc,
  hyperlinkDoc,
  inlineEntityDoc,
  invalidMarksDoc,
  invalidTypeDoc,
  marksDoc,
  olDoc,
  paragraphDoc,
  quoteDoc,
  ulDoc,
} from './documents';

describe('documentToReactTree', function() {
  it('returns null when given an empty document', () => {
    var document = {
      nodeType: BLOCKS.DOCUMENT,
      data: {},
      content: [],
    };
    expect(documentToReactTree(document)).toEqual([]);
  });
  it('renders nodes with default node renderer', function() {
    var docs = [
      paragraphDoc,
      headingDoc(BLOCKS.HEADING_1),
      headingDoc(BLOCKS.HEADING_2),
    ];
    docs.forEach(function(doc) {
      expect(documentToReactTree(doc)).toMatchSnapshot();
    });
  });
  it('renders marks with default mark renderer', function() {
    var docs = [
      marksDoc(MARKS.ITALIC),
      marksDoc(MARKS.BOLD),
      marksDoc(MARKS.UNDERLINE),
      marksDoc(MARKS.CODE),
    ];
    docs.forEach(function(doc) {
      expect(documentToReactTree(doc)).toMatchSnapshot();
    });
  });
  it('renders nodes with passed custom node renderer', function() {
    var options = {
      renderNode: {
        [BLOCKS.PARAGRAPH]: (node, children, defautKey) => (
          <p key={defautKey}>{children}</p>
        ),
      },
    };
    var document = paragraphDoc;
    expect(documentToReactTree(document, options)).toMatchSnapshot();
  });
  it('renders marks with the passed custom mark rendered', function() {
    var options = {
      renderMark: {
        [MARKS.UNDERLINE]: text => <u>{text}</u>,
      },
    };
    var document = marksDoc(MARKS.UNDERLINE);
    expect(documentToReactTree(document, options)).toMatchSnapshot();
  });
  it('does not render unrecognized marks', function() {
    var document = invalidMarksDoc;
    expect(documentToReactTree(document)).toMatchSnapshot();
  });
  it('renders empty node if type is not recognized', function() {
    var document = invalidTypeDoc;
    expect(documentToReactTree(document)).toMatchSnapshot();
  });
  it('renders default entry link block', function() {
    var entrySys = {
      sys: {
        id: '9mpxT4zsRi6Iwukey8KeM',
        link: 'Link',
        linkType: 'Entry',
      },
    };
    var document = embeddedEntryDoc(entrySys);
    expect(documentToReactTree(document)).toMatchSnapshot();
  });
  it('renders ordered lists', function() {
    var document = olDoc;
    expect(documentToReactTree(document)).toMatchSnapshot()
  });
  it('renders unordered lists', function() {
    var document = ulDoc;
    expect(documentToReactTree(document)).toMatchSnapshot()
  });
  it('renders blockquotes', function() {
    var document = quoteDoc;
    expect(documentToReactTree(document)).toMatchSnapshot()
  });
  it('renders horizontal rule', function() {
    var document = hrDoc;
    expect(documentToReactTree(document)).toMatchSnapshot()
  });
  it('does not crash with inline elements (e.g. hyperlink)', function() {
    var document = hyperlinkDoc;
    expect(documentToReactTree(document)).toBeTruthy();
  });
  it('renders hyperlink', function() {
    var document = hyperlinkDoc;
    expect(documentToReactTree(document)).toMatchSnapshot()
  });
  it('renders asset hyperlink', function() {
    var asset = {
      target: {
        sys: {
          id: '9mpxT4zsRi6Iwukey8KeM',
          link: 'Link',
          type: 'Asset',
        },
      },
    };
    var document = inlineEntityDoc(asset, INLINES.ASSET_HYPERLINK);
    expect(documentToReactTree(document)).toMatchSnapshot()
  });
  it('renders entry hyperlink', function() {
    var entry = {
      target: {
        sys: {
          id: '9mpxT4zsRi6Iwukey8KeM',
          link: 'Link',
          type: 'Entry',
        },
      },
    };
    var document = inlineEntityDoc(entry, INLINES.ENTRY_HYPERLINK);
    expect(documentToReactTree(document)).toMatchSnapshot()
  });
  it('renders embedded entry', function() {
    var entry = {
      target: {
        sys: {
          id: '9mpxT4zsRi6Iwukey8KeM',
          link: 'Link',
          type: 'Entry',
        },
      },
    };
    var document = inlineEntityDoc(entry, INLINES.EMBEDDED_ENTRY);
    expect(documentToReactTree(document)).toMatchSnapshot()
  });
});
