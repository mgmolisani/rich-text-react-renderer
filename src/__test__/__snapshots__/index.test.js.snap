// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`documentToReactTree does not render unrecognized marks 1`] = `
Array [
  <p>
    Hello world!
</p>,
]
`;

exports[`documentToReactTree renders asset hyperlink 1`] = `
Array [
  <p>
    
    <span>
        type: 
        asset-hyperlink
         id: 
        9mpxT4zsRi6Iwukey8KeM
    </span>
    
</p>,
]
`;

exports[`documentToReactTree renders blockquotes 1`] = `
Array [
  <p>
    hello
</p>,
  <blockquote>
    world
</blockquote>,
]
`;

exports[`documentToReactTree renders default entry link block 1`] = `
Array [
  <div>
    
</div>,
]
`;

exports[`documentToReactTree renders embedded entry 1`] = `
Array [
  <p>
    
    <span>
        type: 
        embedded-entry-inline
         id: 
        9mpxT4zsRi6Iwukey8KeM
    </span>
    
</p>,
]
`;

exports[`documentToReactTree renders empty node if type is not recognized 1`] = `
Array [
  null,
]
`;

exports[`documentToReactTree renders entry hyperlink 1`] = `
Array [
  <p>
    
    <span>
        type: 
        entry-hyperlink
         id: 
        9mpxT4zsRi6Iwukey8KeM
    </span>
    
</p>,
]
`;

exports[`documentToReactTree renders horizontal rule 1`] = `
Array [
  <p>
    hello world
</p>,
  <hr />,
  <p>
    
</p>,
]
`;

exports[`documentToReactTree renders hyperlink 1`] = `
Array [
  <p>
    Some text 
    <a
        href="https://url.org"
    >
        link
    </a>
     text.
</p>,
]
`;

exports[`documentToReactTree renders marks with default mark renderer 1`] = `
Array [
  <p>
    <i>
        hello world
    </i>
</p>,
]
`;

exports[`documentToReactTree renders marks with default mark renderer 2`] = `
Array [
  <p>
    <b>
        hello world
    </b>
</p>,
]
`;

exports[`documentToReactTree renders marks with default mark renderer 3`] = `
Array [
  <p>
    <u>
        hello world
    </u>
</p>,
]
`;

exports[`documentToReactTree renders marks with default mark renderer 4`] = `
Array [
  <p>
    <code>
        hello world
    </code>
</p>,
]
`;

exports[`documentToReactTree renders marks with the passed custom mark rendered 1`] = `
Array [
  <p>
    <u>
        hello world
    </u>
</p>,
]
`;

exports[`documentToReactTree renders nodes with default node renderer 1`] = `
Array [
  <p>
    hello world
</p>,
]
`;

exports[`documentToReactTree renders nodes with default node renderer 2`] = `
Array [
  <h1>
    hello world
</h1>,
]
`;

exports[`documentToReactTree renders nodes with default node renderer 3`] = `
Array [
  <h2>
    hello world
</h2>,
]
`;

exports[`documentToReactTree renders nodes with passed custom node renderer 1`] = `
Array [
  <p>
    hello world
</p>,
]
`;

exports[`documentToReactTree renders ordered lists 1`] = `
Array [
  <ol>
    <li>
        <p>
            Hello
        </p>
    </li>
    <li>
        <p>
            world
        </p>
    </li>
</ol>,
  <p>
    
</p>,
]
`;

exports[`documentToReactTree renders unordered lists 1`] = `
Array [
  <ul>
    <li>
        <p>
            Hello
        </p>
    </li>
    <li>
        <p>
            world
        </p>
    </li>
</ul>,
  <p>
    
</p>,
]
`;
